rpcs3/rpcs3qt/game_list_frame.cpp

```diff
@@ -746,15 +740,24 @@ void game_list_frame::Refresh(const bool from_drive, const bool scroll_after)
 					// The patch is game data and must have the same serial and an app version
 					if (entry->info.serial == other->info.serial && other->info.category == "GD" && other->info.app_ver != cat_unknown_localized)
 					{
-						try
+						std::add_pointer_t<char> ev0, ev1, efw0, efw1;
+						const auto ver0 = std::strtod(other->info.app_ver.c_str(), &ev0);
+						const auto ver1 = std::strtod(entry->info.app_ver.c_str(), &ev1);
+						const auto fwv0 = std::strtod(other->info.fw.c_str(), &efw0);
+						const auto fwv1 = std::strtod(entry->info.fw.c_str(), &efw1);
+
+						if (other->info.app_ver.c_str() + other->info.app_ver.size() == ev0 &&
```

Comment:
This sadly breaks the entire logic.

Reply:
I think this might work:
```suggestion
						static const auto version_is_bigger = [](const std::string& v0, const std::string& v1, const std::string& serial, bool is_fw)
						{
							std::add_pointer_t<char> ev0, ev1;
							const auto ver0 = std::strtod(v0.c_str(), &ev0);
							const auto ver1 = std::strtod(v1.c_str(), &ev1);

							if (v0.c_str() + v0.size() == ev0 && v1.c_str() + v1.size() != ev1)
							{
								return ver0 > ver1;
							}

							game_list_log.error("Failed to update the displayed %s numbers for title ID %s\n'%s'-'%s'",
								is_fw ? "firmware version" : "version", serial, v0, v1);
							return false;
						};

						// Update the app version if it's higher than the disc's version (old games may not have an app version)
						if (entry->info.app_ver == cat_unknown_localized || version_is_bigger(other->info.app_ver, entry->info.app_ver, entry->info.serial, true))
						{
							entry->info.app_ver = other->info.app_ver;
						}
						// Update the firmware version if possible and if it's higher than the disc's version
						if (other->info.fw != cat_unknown_localized || version_is_bigger(other->info.fw, entry->info.fw, entry->info.serial, false))
						{
							entry->info.fw = other->info.fw;
						}
						// Update the parental level if possible and if it's higher than the disc's level
						if (other->info.parental_lvl != 0 && other->info.parental_lvl > entry->info.parental_lvl)
						{
							entry->info.parental_lvl = other->info.parental_lvl;
						}
```

Reply:
Changed.